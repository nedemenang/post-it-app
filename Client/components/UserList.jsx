import React, { Component } from 'react';
import { List, Subheader } from 'material-ui';
import lodash from 'lodash';
import '../public/style.scss';
import User from './User';


/**
 * @class UserList
 * @extends {Component}
 */
class UserList extends Component {

  /**
   * Creates an instance of UserList.
   * @param {object} props prop object
   * @return {void} return void
   * @memberof UserList
   */
  constructor(props) {
    super(props);
    this.state = {};
  }

  /**
   *
   * Renders user list page
   * @returns {JSX} returns userlist page
   * @memberof UserList
   */
  render() {
    let groupsname = '';
    if (!lodash.isEmpty(this.props.selectedGroup)) {
      groupsname = this.props.selectedGroup.groupname;
    }
    return (
      <div>
        <List>
        <Subheader><strong>Click to add user to group</strong></Subheader>
          {
            this.props.users.map((user, i) => <User
            selectedGroup={this.props.selectedGroup}
            user={user} key={i} />)
          }
          </List>
      </div>
    );
  }
}

export default UserList;
