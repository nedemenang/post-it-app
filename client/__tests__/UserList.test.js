import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserList from '../components/UserList.jsx';

describe('the User list component', () => {
  let mountedComponent;

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: `${str2}static/files/blank-profile-pic.png`
  };

  const users = [{
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    profilePic: `${str2}static/files/blank-profile-pic.png` },
  {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    profilePic: `${str2}static/files/blank-profile-pic.png` }];

  const group = { groupId: 'TUsUaucS5nb5kuOP6SpvTUcX9v3',
    groupName: 'Test Group',
    newMessage: false };

  const userListItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<UserList users = {users}
      selectedGroup={group} loggedInUser = {loggedInUser}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = userListItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = userListItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });


  it('should recieve props', () => {
    expect(Object.keys(userListItem().props()).length).toBeGreaterThan(0);
  });
});
