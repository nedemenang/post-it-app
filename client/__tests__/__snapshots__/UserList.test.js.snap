// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the User list component should match snapshot test 1`] = `
<div>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search for Users"
    type="text"
    value=""
  />
  <List>
    <Subheader
      inset={false}
    >
      <strong>
        Click to add user to group
      </strong>
    </Subheader>
  </List>
</div>
`;
